test/optimizer/analyze_test.cpp

```diff
@@ -0,0 +1,118 @@
+#include <limits>
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "main/db_main.h"
+#include "optimizer/analyze.h"
+#include "optimizer/statistics/column_stats.h"
+#include "storage/garbage_collector_thread.h"
+#include "storage/projected_row.h"
+#include "storage/sql_table.h"
+#include "test_util/catalog_test_util.h"
+#include "test_util/test_harness.h"
+#include "transaction/transaction_context.h"
+#include "transaction/transaction_manager.h"
+#include "type/type_id.h"
+#include "type/type_util.h"
+
+namespace terrier::storage::index {
+
+class AnalyzeTests : public TerrierTest {
+ private:
+  catalog::IndexSchema unique_schema_;
+  catalog::IndexSchema default_schema_;
+
+ public:
+  std::unique_ptr<DBMain> db_main_;
+  common::ManagedPointer<transaction::TransactionManager> txn_manager_;
+
+  // SqlTable
+  catalog::Schema table_schema_;
+  storage::SqlTable *sql_table_;
+  storage::ProjectedRowInitializer tuple_initializer_ =
+      storage::ProjectedRowInitializer::Create(std::vector<uint16_t>{1}, std::vector<uint16_t>{1});
+
+ protected:
+  void SetUp() override {
+    db_main_ = terrier::DBMain::Builder().SetUseGC(true).SetUseGCThread(true).SetRecordBufferSegmentSize(1e6).Build();
+    txn_manager_ = db_main_->GetTransactionLayer()->GetTransactionManager();
+
+    auto col = catalog::Schema::Column(
+        "attribute", type::TypeId::INTEGER, false,
+        parser::ConstantValueExpression(type::TransientValueFactory::GetNull(type::TypeId::INTEGER)));
```

Comment:
This test is a good first start, but I would add varchars + floats to make sure that we support these types.